src/state/state.js

```diff
@@ -188,14 +188,14 @@
          * @ignore
          */
         function _renderFrame(time) {
-            // update all game objects
-            me.game.update(time);
-            // render all game objects
-            me.game.draw();
             // schedule the next frame update
             if (_animFrameId !== -1) {
                 _animFrameId = window.requestAnimationFrame(_renderFrame);
             }
+            // update all game objects
+            me.game.update(time);
+            // render all game objects
+            me.game.draw();
```

Comment:
It's asynchronous by the nature of video-sync. :) The callback is queued for the next available frame. In this context, "available" just means that no other code is currently running.

Imagine you had a lot of complex AI and particles, and your frame processing was taking 18ms. With an RAF callback triggered at T+0, the frame processing goes beyond the 16.667ms timeframe you have for T+1. So the UA ends up dropping a frame, and the next RAF callback fires at T+2, regardless of what point during that 18ms processing time the RAF was scheduled; it could have happened before or after T+1, and the result will still be the same. It drops a frame at T+1 because JavaScript is currently running at that time.

This is what makes it better than a timer (or interval) for animations; the operation of RAF is deterministic.